lib/eblox/data/provider.ex

```diff
@@ -53,33 +58,34 @@ defmodule Eblox.Data.Provider do
   def perform(:died, _id, payload), do: {:transition, :*, payload}
   def perform(_state, _id, payload), do: {:transition, :scan, payload}
 
-  @spec handle_changes(t()) :: [:ok]
-  def handle_changes(%Provider{created: _, deleted: _, changed: _} = changes) do
+  @spec handle_changes(module(), t()) :: [:ok]
+  def handle_changes(impl, %Provider{created: _, deleted: _, changed: _} = changes) do
     changes
     |> Flow.from_enumerable()
     |> Flow.flat_map(fn {action, list} -> Enum.map(list, &{action, &1}) end)
     |> Flow.partition()
-    |> Flow.reduce(fn -> [] end, fn {action, elem}, acc -> [action(action, elem) | acc] end)
+    |> Flow.reduce(fn -> [] end, fn {action, elem}, acc -> [action(impl, action, elem) | acc] end)
     |> Enum.to_list()
   end
 
   @interval Application.compile_env(:eblox, :parse_interval, 10)
 
-  @spec action(:created | :deleted | :changed, binary()) :: :ok
-  defp action(:created, file) do
-    with {:ok, _server} <-
-           Siblings.start_child(Eblox.Data.Post, file, %{file: file},
+  @spec action(module(), :created | :deleted | :changed, binary()) :: :ok
+  defp action(impl, :created, file) do
+    with properties = %{} <- impl.initial_payload(file),
```

Comment:
I kinda dislike how we silently omit any possible issue here. Any idea how to do it better? `Logger` maybe?

Reply:
I added error logging to this function and type specs for possible errors.